### PR TITLE
Test "genYAML" to generate the yaml file for aero DA

### PR DESCRIPTION
**Pull request by andytangborn**
This PR adds a test to build an aero DA yaml, following the form of the land DA yaml test.

### CODE REVIEW

File: ush/ufsda/genYAML.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def get_runtime_config(config_dict):
     bkg_tstep = str(config_dict.get('BKG_TSTEP', f"PT{assim_freq}H"))
     component_dict = {
         'atmos': 'ATM',
-        'chem': 'AERO',
```
Comment (CoryMartin-NOAA) [inline ush/ufsda/genYAML.py]: this should stay chem because that is the COMROT structure for NCO.

File: ush/ufsda/yamltools.py [hunk 2]
```diff
@@ -132,6 +136,35 @@ def fv3land_geom_dict(case, levs, ntiles, layout, io_layout):
     }
     return outdict
 
+def fv3aero_geom_dict(case, levs, ntiles, layout, io_layout):
```
Comment (CoryMartin-NOAA) [inline ush/ufsda/yamltools.py:140]: what is different here besides the fieldmetadata file? I don't think we need the 'time invariant' part, right?

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -175,4 +189,6 @@ endif()
 add_subdirectory(atm)
 
 # gdas aerosol tests
-add_subdirectory(aero)
+if (${BUILD_GDASBUNDLE})
```
Comment (CoryMartin-NOAA) [inline test/CMakeLists.txt]: why this change in this PR?

**Comment by CoryMartin-NOAA** [comment]
@andytangborn a few comments on the code, plus all the tests are failing...

**Comment by andytangborn** [comment]
Ok, but it failed when it was "chem" it gave a key error "aero".  So that
wasn't the right fix then.

File: ush/ufsda/genYAML.py (hunk 1 above)
Comment (andytangborn) [reply ush/ufsda/genYAML.py]: I'm getting a key error "aero" otherwise. Changing this to "aero" fixed it. So I need another solution.

File: ush/ufsda/yamltools.py (hunk 2 above)
Comment (andytangborn) [reply ush/ufsda/yamltools.py:140]: I wasn't sure. I can take it out.

File: test/CMakeLists.txt (hunk 3 above)
Comment (andytangborn) [reply test/CMakeLists.txt]: If it's not needed, I can remove it.

File: ush/ufsda/yamltools.py (hunk 2 above)
Comment (jiaruidong2017) [reply ush/ufsda/yamltools.py:140]: Yes, the `time invariant` part is land specific.

File: ush/ufsda/genYAML.py (hunk 1 above)
Comment (jiaruidong2017) [reply ush/ufsda/genYAML.py]: In `test/genYAML_prep_aero.sh`, try to replace `COMPONENT: aero` with `COMPONENT: chem`.

Comment (CoryMartin-NOAA) [reply ush/ufsda/genYAML.py]: in `test/genYAML_prep_aero.sh` what if you change it to `chem`? Does it still error? The directory structure will be like `gdas.20230303/12/chem` so we need it to be chem.

File: test/CMakeLists.txt (hunk 3 above)
Comment (CoryMartin-NOAA) [reply test/CMakeLists.txt]: It's mainly that if you aren't doing anything to `test/aero` then we shouldn't change that `add_subdirectory` part right now

File: test/genYAML_prep_aero.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  layout_x: '8'
+  layout_y: '8'
+  OBS_LIST: ${srcdir}/parm/aero/obs/lists/gdas_aero_prototype.yaml
+  OPREFIX: gdas.t06z.
+  CDATE: '2019061400'
+  PDY: '20190614'
+  cyc: '06'
+  valid_time: '2019-06-14T06:00:00Z'
+  aero_window_length: PT6H
+  CASE: 'C48'
```
Comment (CoryMartin-NOAA) [inline test/genYAML_prep_aero.sh]: isn't this C96 in your sample case?

Comment (andytangborn) [reply test/genYAML_prep_aero.sh]: Yes. I'll change this.

Comment (andytangborn) [reply test/genYAML_prep_aero.sh]: There's also a CASE ENKF 48 below that. Do we even need that since this is for 3dvar?

Comment (CoryMartin-NOAA) [reply test/genYAML_prep_aero.sh]: not really, although that is often (probably incorrectly) used as the analysis resolution

File: test/CMakeLists.txt (hunk 3 above)
Comment (CoryMartin-NOAA) [reply test/CMakeLists.txt]: Yeah I would revert this change in this PR, just so that we only touch it later when we add 3dvar or hofx tests

Comment (andytangborn) [reply test/CMakeLists.txt]: done.

**Review by jiaruidong2017 — APPROVED** [review]

**Review by CoryMartin-NOAA — APPROVED** [review]
